tests/Doctrine/Tests/Common/Cache/CacheTest.php

```diff
@@ -29,6 +29,21 @@ public function testBasicCrudOperations($value)
         $this->assertFalse($cache->contains('key'));
     }
 
+    public function testCacheKeyIsCaseSensitive()
+    {
+        $cache = $this->_getCacheDriver();
+
+        $this->assertTrue($cache->save('key', 'value'));
+        $this->assertTrue($cache->contains('key'));
+        $this->assertSame('value', $cache->fetch('key'));
+
+        $this->assertFalse($cache->contains('KEY'));
+        $this->assertFalse($cache->fetch('KEY'));
+
+        $cache->delete('KEY');
+        $this->assertTrue($cache->contains('key', 'Deleting cache item with different case must not affect other cache item'));
```

Comment:
This test passes also on case-insensitive operating systems because of the hash being different for different case keys: tricky :-\